editor/modes/visual-editor/index.js

```diff
@@ -52,9 +53,20 @@ class VisualEditor extends Component {
 	}
 
 	selectAll( event ) {
-		const { uids } = this.props;
+		const { uids, onMultiSelect } = this.props;
+		event.preventDefault();
+		onMultiSelect( first( uids ), last( uids ) );
+	}
+
+	undoOrRedo( event ) {
+		const { onRedo, onUndo } = this.props;
+		if ( event.shiftKey ) {
+			onRedo();
+		} else {
+			onUndo();
+		}
+
```

Comment:
>What is the purpose of the prevent default here?

It ensures that the browser default undo behavior is overridden by our custom implementation.

It's likely browser specific, but at least in Chrome I notice some differences in how undo stacks are generated with `contenteditable` vs. our save intervals with Editable changes.